File: bs-inventory/app/models/book.py
import app


class Book(app.db.Model):
    id = app.db.Column(app.db.Integer, primary_key=True)
    title = app.db.Column(app.db.String(255), nullable=False)
    author = app.db.Column(app.db.String(255), nullable=False)
    published_date = app.db.Column(app.db.String(255), nullable=True)

    def __repr__(self):
        return "<Book %r>" % self.title

    def to_dict(self):
        return {
            column.name: getattr(self, column.name) for column in self.__table__.columns
        }

File: bookstore-api/src/models/book.js
const { Model } = require('sequelize');
const models = require('.');
const Author = require('./author');

class Book extends Model {}

Book.init(
  {
    id: {
      type: models.Sequelize.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    title: {
      type: models.Sequelize.STRING,
      allowNull: false,
    },
    published_date: {
      type: models.Sequelize.STRING,
      allowNull: true,
    },
    isbn: {
      type: models.Sequelize.STRING,
      allowNull: true,
    },
    quantity_on_hand: {
      type: models.Sequelize.INTEGER,
      allowNull: false,
      defaultValue: 0,
    },
  },
  {
    sequelize: models.sequelize,
    modelName: 'book',
  },
);

Book.belongsTo(Author);
Author.hasMany(Book);

module.exports = Book;
